warehouse_simulation_script.py

```diff
@@ -0,0 +1,489 @@
+            
+            # Calculate distance to target
+            dx = target_pos.x - current_pos.x
+            dy = target_pos.y - current_pos.y
+            if np.sqrt(dx**2 + dy**2) < 0.5:
+                logger.info("\tDingo reached target position")
+                target_reached = True
+            else:
+                cmd_vel.linear.x = 0.5
+                dingo_cmd_vel_pub.publish(cmd_vel)
+
+        logger.info("\tMoving UR10 joints")
+        move_ur10_joints(node, loop_iteration, sim_backend)
+        time.sleep(1.5)
+
+
+def yaw_to_quaternion(yaw):
+    """Convert a yaw angle (in radians) to a geometry_msgs.msg.Quaternion.
+
+    Returns a Quaternion message with normalized components (w, x, y, z).
+    """
+    yaw = np.arctan2(np.sin(yaw), np.cos(yaw))
+    half_yaw = yaw / 2.0
+    w = float(np.cos(half_yaw))
+    x = 0.0
+    y = 0.0
+    z = float(np.sin(half_yaw))
+    # Normalize to be safe
+    norm = float(np.sqrt(w * w + x * x + y * y + z * z))
+    if norm == 0.0:
+        norm = 1.0
+    q = Quaternion(w=w / norm, x=x / norm, y=y / norm, z=z / norm)
+    return q
+
+# ************************************
+# Main loop
+# ************************************
+
+
+def main():
+    # Map simulation differences
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--sim-backend", choices=["isaacsim", "o3de", "gazebo"], default="isaacsim",
+                        help="Choose which asset backend to use (isaacsim, o3de or gazebo).")
+    args, unknown = parser.parse_known_args()
+
+    logging.basicConfig(level=logging.INFO, format='%(levelname)s: %(message)s')
+    logging.info("Starting warehouse simulation script")
+
+    # Configure simulation backend (sets global variables)
+    configure_simulation_backend(args.sim_backend)
+
+    # Initialize ROS 2 node and client library
+    rclpy.init()
+    node = rclpy.create_node("simulation_interfaces_demo")
+    
+    # 1. Get features
+    features = get_features(node, GET_FEATURES_SERVICE)
+    if features is None:
+        return
+    
+    # 1.1 Load world
+    logging.info("1.1 Loading world (if supported)")
+    if SimulatorFeatures.WORLD_LOADING in features.features:
+        load_world(node, LOAD_WORLD_SERVICE, WORLD_URI)
+        time.sleep(1.5)
+
+    # 1.2 Spawn and despawn object
+    logging.info("1.2 Spawning and despawning a table")
+    initial_pose = PoseStamped()
+    initial_pose.pose.position = Point(x=0.0, y=0.0, z=1.19)
+    spawn_entity(node, SPAWN_ENTITY_SERVICE, TABLE_URI, format_entity_name("Table", RENAME_ENTITY), initial_pose)
+    time.sleep(1.5)
+    delete_entity(node, DELETE_ENTITY_SERVICE, format_entity_name("Table", RENAME_ENTITY))
+    time.sleep(1.5)
+
+    # 1.3 Spawn Table, some boxes, Dingo and UR10 with helper method
+    logging.info("1.3 Spawning the full scene")
+    spawn_scene(node, SPAWN_ENTITY_SERVICE)
+    
+    # Move the robot (this makes play/pause more interesting)
```

Comment:
I think this section is new behavior that does not exist in the previous version of the script in https://github.com/ros-simulation/simulation_interfaces_examples/pull/2